src/Renderer/LayeredMaterial.js

```diff
@@ -244,83 +283,87 @@ LayeredMaterial.prototype.setTexture = function(texture, layer, slot, pitScale)
     // BEWARE: array [] -> size: 0; array [10]="wao" -> size: 11
     this.Textures[layer][slot] = texture ? texture : emptyTexture;
     this.pitScale[layer][slot] = pitScale ? pitScale : new THREE.Vector3(0.0, 0.0, 1.0);
+};
 
-    // TEMP
-    if (texture === null) {
-        var l = this.slotToLayerIndex(slot);
-        if (l >= 0) {
-            this.paramLayers[l].z = 0;
-        }
+LayeredMaterial.prototype.setColorLayerParameters = function(params) {
+    for (var l = 0; l < params.length; l++) {
```

Comment:
`setColorLayerParameters` is supposed to be used only once. So maybe add a few safety checks at the beginning:
* `loaded` should be false
*  `params.length` should be 0